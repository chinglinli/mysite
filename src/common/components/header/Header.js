import React from 'react'
import PropTypes from 'prop-types'
import {Link} from 'react-router'

import styles from './Header.scss'


export default class Header extends React.Component {
  static PropTypes = {
    siteName: PropTypes.string.isRequired
  }

  render() {
    return (
      <div className={styles.root}>
        <div className={styles.logo}/>
        <div className={styles.title}><Link className={styles.link} to='/'>{this.props.siteName}</Link></div>
      </div>
    )
  }
}
