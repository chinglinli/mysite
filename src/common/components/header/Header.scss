@import '../../css/base.scss';

.root {
  background: $dominant-color;
  height: 5.625rem;
  display: flex;

  .logo {
    background: url(../../img/logo.png) no-repeat center;
    margin-top: 1rem;
    margin-left: 1rem;

    width: 3.625rem;
    height: 3.625rem;

    transform: rotate(-10deg);
    transition-duration: 0.8s;
    -webkit-backface-visibility: hidden;

    &:hover {
      transform: rotate(10deg);
    }
  }

  .title {
    flex-grow: 1;

    display: flex;
    align-items: center;
    margin-left: 2rem;

    color: white;
    font-size: 1.3rem;
    font-weight: 700;

    .link {
      @extend .link;
      color: white;
    }
  }

  @media (min-width: 900px) {
    .title {
      font-size: 1.5rem;
    }
  }

  @media (min-width: 1200px) {
    .title {
      font-size: 2rem;
    }
  }
}



