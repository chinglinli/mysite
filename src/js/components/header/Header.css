@value dominant-color from '../../css/colors.css';

.root {
    background: dominant-color;
}

.logo {
    width: 58px;
    height: 58px;

    transform: rotate(-10deg);
    transition-duration: 0.8s;
}

.logo:hover {
     transform: rotate(10deg);
}

.title {
    color: white;
    font-size: 2em;
    text-decoration: none;
}
