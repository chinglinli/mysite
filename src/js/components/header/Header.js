import * as React from 'react';
import {Link} from 'react-router';

import styles from './Header.scss';


export default class Header extends React.Component {
  static PropTypes = {
    siteName: React.PropTypes.string.isRequired
  };

  render() {
    return (
      <div className={styles.root}>
        <div className={styles.logo}/>
        <div className={styles.title}><Link className={styles.link} to='/'>{this.props.siteName}</Link></div>
      </div>
    );
  }
}
