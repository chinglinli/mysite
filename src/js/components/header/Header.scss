@import '../../../css/colors.scss';

.root {
    background: $dominant-color;
    height: 85px;

    .logo {
        float: left;
        background: url(../../img/logo.png) no-repeat center;
        margin-top: 10px;
        margin-left: 20px;

        width: 58px;
        height: 58px;

        border: .1em solid #FFF;
        box-shadow: 0 1px 4px rgba(0,0,0,0.15);

        transform: rotate(-10deg);
        transition-duration: 0.8s;

        &:hover {
            transform: rotate(10deg);
        }
    }

    .title {
        display: inline-block;
        margin-left: 25px;
        margin-top: 20px;

        color: white;
        font-size: 2em;
        font-weight: 700;
    }
}



