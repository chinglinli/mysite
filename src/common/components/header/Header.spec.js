import React from 'react'
import renderer from 'react-test-renderer'

import Header from './Header'


test('It should render Header component correctly', () => {
  const siteName = 'Site Name'

  const component = renderer.create(
    <Header siteName={siteName}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
