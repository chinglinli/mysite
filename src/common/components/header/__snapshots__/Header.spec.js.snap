// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Header component correctly 1`] = `
<div
  className="root"
>
  <div
    className="logo"
  />
  <div
    className="title"
  >
    <a
      className="link"
      onClick={[Function]}
      style={Object {}}
    >
      Site Name
    </a>
  </div>
</div>
`;
